### PR TITLE
Geographical map: Show network links between locations

### PR DESCRIPTION
**Pull request by kedare**
This PR add an optional links view on the globe map, it take the links from the discovery protocols and use the location attached to device to build them.

It's a first version, I'm sure there are some improvements that could be done :)
(For example, I could not make successfully the SQL request using joins instead of all the mapping in the where, or make the PolyLine to cross the border of the maps (for example to link APAC to US without having the link crossing Europe instead of just going between them, but I could not find any leaflet setting to do this, not sure it is possible)

Screenshot 2020-03-10 at 16 45 47

It is using the same rule than on the network map (for color and link size)

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by SourceDoctor** [comment]
could you change default value to false

and make it changeable via global Settings?
file: resources/lang/en/settings.php

i would position it here:
Global Settings -> external -> Location Settings

File: includes/html/common/worldmap.inc.php [hunk 1]
```diff
@@ -130,6 +130,123 @@
 marker.bindPopup(title);
     markers.addLayer(marker);\n";
     }
+
+    if (Config::get('network_map_show_on_worldmap', true)) {
```
Comment (SourceDoctor) [inline includes/html/common/worldmap.inc.php]: , true   is not needed because default value is defined via config_definitions file

**Comment by kedare** [comment]
Fixed, I added a new part in the settings related to the world map as there are probably more settings coming (and it's not something external)

However I could not get the l18n to work correctly. (See screenshot), am I doing something wrong ? Any idea ?

Screenshot 2020-03-10 at 19 25 16

**Comment by kedare** [comment]
Fixed with `./lnms translation:generate`

**Comment by PipoCanaja** [comment]
@kedare 
Seems that Travis detects an issue in the config json file : 
```
1) LibreNMS\Tests\YamlTest::testConfigSchema
config_definitions.json does not validate. Violations:
[config.network_map_show_on_worldmap.order] The property order is required
```

**Comment by kedare** [comment]
Should be fixed now :)

**Comment by kedare** [comment]
Looks like the failing CI is related to something else ?

**Comment by SourceDoctor** [comment]
@kedare 
crash causes an OS test. Will be fixes soon

**Comment by PipoCanaja** [comment]
@kedare Bonjour. Tests are fixed now. Bye

**Comment by kedare** [comment]
Great, thanks :)

**Comment by SourceDoctor** [comment]
@kedare please resolve merge conflict

**Comment by kedare** [comment]
Should be ok now

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-62-release-changelog-march-2020/11423/1

**Comment by bernierherve** [comment]
Hi,
Very good functionnality.
But it works only on Overview>>Maps>>Geographical.
Could be a good idea to implement in WorldMap gidget also
Thanks in advance

**Pull request merged**